patchwork-capabilities/src/main/java/net/minecraftforge/common/capabilities/Capability.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.capabilities;
+
+import java.util.concurrent.Callable;
+
+import com.google.common.base.Throwables;
+
+import net.minecraft.nbt.Tag;
+import net.minecraft.util.math.Direction;
+import net.minecraftforge.common.util.LazyOptional;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * This is the core holder object Capabilities.
+ * Each capability will have ONE instance of this class,
+ * and it will the the one passed into the ICapabilityProvider functions.
+ *
+ * The CapabilityManager is in charge of creating this class.
+ */
+public class Capability<T>
+{
+    public static interface IStorage<T>
+    {
+        /**
+         * Serialize the capability instance to a NBTTag.
+         * This allows for a central implementation of saving the data.
+         *
+         * It is important to note that it is up to the API defining
+         * the capability what requirements the 'instance' value must have.
+         *
+         * Due to the possibility of manipulating internal data, some
+         * implementations MAY require that the 'instance' be an instance
+         * of the 'default' implementation.
+         *
+         * Review the API docs for more info.
+         *
+         * @param capability The Capability being stored.
```

Comment:
Putting a link here is useless, the JavaDoc already would include the method parameter types

Reply:
Links are still useful, and they allow the IDE to provide better error reporting. They're more useful with Minecraft classes, since they provide the potential to be remapped.